staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	flowcontrol "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// The objects that define an apiserver's initial behavior
+var (
+	PreservedPriorityLevelConfigurations = []*flowcontrol.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		DefaultPriorityLevelConfigurationDefault,
+	}
+	PreservedFlowSchemas = []*flowcontrol.FlowSchema{
+		DefaultFlowSchemaExempt,
+		DefaultFlowSchemaDefault,
+	}
+)
+
+// Initial PriorityLevelConfiguration objects
+var (
+	DefaultPriorityLevelConfigurationExempt = newPriorityLevelConfiguration(
+		flowcontrol.PriorityLevelConfigurationNameExempt,
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementExempt,
+		},
+	)
+	DefaultPriorityLevelConfigurationDefault = newPriorityLevelConfiguration(
+		"default", // the "catch-all" priority-level
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementLimited,
+			Limited: &flowcontrol.LimitedPriorityLevelConfiguration{
+				AssuredConcurrencyShares: 100,
```

Comment:
do you have a better number for recommendation?

Reply:
or preventing users from modifying the catch-all PL is probably not a good idea, just protecting the exempt would be sufficient?

Reply:
I don't think we have a choice, since a) the enforcement code pretends this exists and b) we have no way of knowing if a proposed modification is "reasonable" or not, since we do probably need this to remain reasonable.

What if we set this to 1 to have the smallest impact possible on the shares? Users can make their own catch-all if that doesn't work for them.

Reply:
IIRC we discussed this in our last well-attended meeting, and concurred with the following.  Giving the catch-all level a relatively miniscule share is problematic because it will choke off some traffic; even for unexpected traffic, a miniscule share is damn near equivalent to no share, which I think we agree we do not want.  Giving the catch-all priority level a significant share is a waste if it is normally unused.  But the catch-all priority level does not have to be normally unused.  Remember, it is actually the FlowSchema that "catches".  We favored using the catch-all priority level for some of the normal traffic and giving it a significant share.  Since we are fixing only one "share" quantity, and all the share quantities are relative, the admins can make that fixed share quantity map to any concurrency quantity they desire.

Reply:
> (a) is best achieved by giving catch-all a very small slice, so that throttling happens quickly.

for the record, in the previous implementation, even 1 concurrencyShare doesnt guarantee the flow-schema will get a share. 

my point of giving the "catch-all" a normal share is based on the fact that "practically it's tough for admins to define a set of flow-schemas to cover/match all the traffic proactively". @lavalamp also suggests users to define their own "catch-all" to ease the pain of mis-killing unclassified traffic. if so, i suppose the new "catch-all" should be included in the suggested configuration, b/c most users will be finally adding it to their cluster.  in terms of UX, i'd expect users can be opt-in this feature w/o pain by simply swtiching on the feature-gate, instead of wasting time diving into the flow-schemas to find out which one is problematic.. the suggested+mandatory set should be sufficient for working w/ most clusters and users are not required to take any extra action unless they feel like more enhancements

Reply:
In #85319 and the feature branch before it, the calculation of concurrency from shares and total goes like this:
```
 		plState.qsConfig.ConcurrencyLimit = int(math.Ceil(float64(reqMgr.serverConcurrencyLimit) * float64(plState.config.Limited.AssuredConcurrencyShares) / shareSum))
```

(so a share of even 1 gets a non-zero amount of concurrency).

Reply:
interesting, does that math work? it looks that the sum of actual shares can be greater than the total share

Reply:
As discussed in the meeting, the excess (if any) will be negligible.